### ISSUE TITLE
AlphaEdit fpr mistralai/Mistral-7B-v0.1

### ISSUE THREAD

**Issue by ShidaWang02** [zjunlp/EasyEdit#490]
非常感谢您的优秀工作！
我现在在尝试使用AlphaEdit对Mistral-7B-v0.1进行编辑，hparams参考了MEMIT中Mistral-7B-v0.1的参数：

{
    "model_name": "Mistral-7B",
    "layers": [4, 5, 6, 7, 8],
    "clamp_norm_factor": 4,
    "layer_selection": "all",
    "fact_token": "subject_last",
    "v_num_grad_steps": 25,
    "v_lr": 5e-1,
    "v_loss_layer": 31,
    "v_weight_decay": 1e-3,
    "kl_factor": 0.0625,
    "mom2_adjustment": true,
    "mom2_update_weight": 15000,
    "rewrite_module_tmp": "model.layers.{}.mlp.down_proj",
    "layer_module_tmp": "model.layers.{}",
    "mlp_module_tmp": "model.layers.{}.mlp",
    "attn_module_tmp": "model.layers.{}.self_attn",
    "ln_f_module": "model.norm",
    "lm_head_module": "lm_head",
    "mom2_dataset": "wikipedia",
    "mom2_n_samples": 100000,
    "mom2_dtype": "float32",
    "nullspace_threshold":2e-2,
    "L2":10
}

但是似乎效果不是很好，在优化 $v^*$的过程中loss下降到某一个不算低的值就不再下降了(或许对LLaMA2-7B也有同样的问题)，请问您有尝试过在这个模型上使用AlphaEdit进行编辑，是否可以提供一些合适的参数或者调整参数的经验呢？

**Comment by JizhanFang** [comment]
你这边能提供一下最终运行得到的结果指标吗？可能更方便我帮助你解决问题

**Comment by ShidaWang02** [comment]
例如，我要执行的编辑为 [MODEL] -> [ Mistral]，下面是运行过程中的loss记录，我做了一些改动，在计算loss时将kl_loss删去了,让v的优化更简单一些，即
`loss = nll_loss.to(model_device) + kl_loss.to(model_device) + weight_decay.to(model_device)`改为
`loss = nll_loss.to(model_device) + weight_decay.to(model_device)`

但最后得到的输出概率也只有0.038，最后的编辑结果也是失败的，和我在其他模型（例如LLaMA3-8B，GPT-J-6B）的表现有比较大的区别。

![Image](https://github.com/user-attachments/assets/0fc37848-4e32-4bc7-9e71-9acb66099e60)

**Comment by JizhanFang** [comment]
个人感觉这里编辑效果不好不是由alphaedit直接有关的超参导致的，也就是和nullspace_threshold和L2无关，你可以尝试调一下clamp_norm_factor、v_lr、v_weight_decay这几个参数。

**Comment by ShidaWang02** [comment]
感谢，我再去尝试一下

**Comment by ShidaWang02** [comment]
我发现是因为LLaMa2-8B和Mistral-7B的tokenizer对于空格" "的编码比较特殊，导致编辑失败。

**Comment by JizhanFang** [comment]
请问你这边有解决这个问题吗？我会尽快复现并且修复这个问题

**Comment by ShidaWang02** [comment]
Llama2-8B和Mistral-7B的tokenizer对于位于句首的空格" "，会将其tokenize成"▁"对应的token_id而不是的"▁▁"对应的token_id（tokenizer(“ ”)），在优化$v$时"▁"的loss很难降得下去，把它删了就好了。
